### PR TITLE
4156, 4151

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:* /bcgov/entity#4156, /bcgov/entity#4151

*Description of changes:*
Fixes and updates to step 1 so form validation errors show when a user resumes an application

*Office Addresses*
- added references to address components (sbc common component BaseAddress) so the form validate() function is accessible
- add checks on mounted to compare the default address state to what is loaded so re-validation can occur to show errors (e.g. non BC province)
- updated formValid() check to fix a bug where the validity for all addresses were not correct

*Business Contact*
- Added a default business contact object for comparison to re-evaluate validation (e.g. invalid email)
- Added check on mount to re-validate form if it is not the same as the default form state

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by ochiu** [comment]
This ended up being a little tricky as the address components are the common components, so I had to add some references to access the validation. We also didn't want errors to show on a fresh form so for step 1 forms I added some checks against what we consider a default state before re-validating.

The other steps don't seem to be as much as issue since we open an edit form and the user hits done.

File: src/components/DefineCompany/OfficeAddresses.vue [hunk 1]
```diff
@@ -90,7 +90,7 @@
               </div>
               <div class="address-wrapper"
                    v-if="!isSame(mailingAddress, deliveryAddress, 'actions') || !inheritMailingAddress">
-                <delivery-address
+                <delivery-address ref='regDeliveryAddress'
```
Comment (severinbeauvais) [inline src/components/DefineCompany/OfficeAddresses.vue:93]: Please use consistent (double) quotes here.

**Comment by ochiu** [comment]
> That's actually a pretty clean solution. Well done!
> 
> Obviously we were missing unit tests to verify various scenarios. Could you please add some to validate your changes (as well as catch any future regressions)? Thanks.

I looked as business contacts and it does have invalid type checks. I am thinking it must have to do more with how the whole app life cycle works and when validation is triggered.

Also, the address components are from the sbc-common-components. I can definitely try to dig into this, just not sure if I should invest the time now as it could be a time sink.

**Comment by severinbeauvais** [comment]
> > Obviously we were missing unit tests to verify various scenarios. Could you please add some to validate your changes (as well as catch any future regressions)? Thanks.
> 
> I looked as business contacts and it does have invalid type checks. I am thinking it must have to do more with how the whole app life cycle works and when validation is triggered.
> Also, the address components are from the sbc-common-components. I can definitely try to dig into this, just not sure if I should invest the time now as it could be a time sink.

It's not the BaseAddress component that I want to test, but the OfficeAddresses component... just some simple tests to make sure that validation is done when the component is mounted, and passes/fails according to the page content.

If you're out of time then just add some "TODO" comments in the appropriate unit test file with details on what still needs to be tested (eg, which validation scenarios). Thx.

**Comment by ochiu** [comment]
> > > Obviously we were missing unit tests to verify various scenarios. Could you please add some to validate your changes (as well as catch any future regressions)? Thanks.
> > 
> > 
> > I looked as business contacts and it does have invalid type checks. I am thinking it must have to do more with how the whole app life cycle works and when validation is triggered.
> > Also, the address components are from the sbc-common-components. I can definitely try to dig into this, just not sure if I should invest the time now as it could be a time sink.
> 
> It's not the BaseAddress component that I want to test, but the OfficeAddresses component... just some simple tests to make sure that validation is done when the component is mounted, and passes/fails according to the page content.
> 
> If you're out of time then just add some "TODO" comments in the appropriate unit test file with details on what still needs to be tested (eg, which validation scenarios). Thx.

Ahh okay, makes sense. Added some tests for office addresses.

**Comment by severinbeauvais** [comment]
Love the unit tests - nice and simple and to the point. Thanks!

**Pull request merged**